### PR TITLE
API 500 Scopes support

### LINKED ISSUE #248: Support to Scope in API500
### Scenario/Intent
Add support to Scope in API500.

### Environment Details
 - **oneview Cookbook Version:** unreleased next version (probably 2.4.0)
 - **OneView API Version:** 500
 - **OneView Version:** 3.10

### PR DESCRIPTION
**Pull request by ricardogpsf**
### Description
Adding support to scopes of API 500

### Issues Resolved
#248 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [ ] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by ricardogpsf** [comment]
~~This code is not ready yet. It's just to other people verify if the way is correct.~~
Now, it is ready.

File: attributes/default.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@
 # A resource provider must be defined for this version. For example, when set to 200, it will look
 # for the resource in OneviewCookbook::API200. When 300, it will look in OneviewCookbook::API300
 # See the libraries/resources directory for more info on supported API versions
-default['oneview']['api_version'] = 200
+default['oneview']['api_version'] = 300
```
Comment (tmiotto) [inline attributes/default.rb]: Don't do it, it is a breaking change

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ## 2.3.1 (Unreleased)
+Adds API 500 support to the following HPE OneView resources:
+  - oneview_scope resource
```
Comment (tmiotto) [inline CHANGELOG.md]: There is no need to say again it is a resource if you're saying in the sentence before they're all resources.

Please list only the resource name, e.g. `- oneview_scope`

File: libraries/oneview_helper.rb [hunk 3]
```diff
@@ -14,6 +14,27 @@
 module OneviewCookbook
   # Helpers for Oneview Resources
   module Helper
+    # Get resource class that matches the type given
+    # @param [String] type Name of the desired class type
+    # @param [Module] api_module Module who the desired class type belongs to
+    # @param [String] variant Variant (C7000 or Synergy)
+    # @return [Class] Resource class or nil if not found
```
Comment (tmiotto) [inline libraries/oneview_helper.rb]: This doc is wrong, it does not returns 'nil' f not found, it raises an error.

File: libraries/resource_providers/api500.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+module OneviewCookbook
+  # Module for Oneview API 500 Resources
+  module API500
+    SUPPORTED_VARIANTS ||= %w[C7000 Synergy].freeze
+
+    # Get resource class that matches the type given
+    # @param [String] type Name of the desired class type
+    # @param [String] variant Variant (C7000 or Synergy)
+    # @return [Class] Resource class or nil if not found
```
Comment (tmiotto) [inline libraries/resource_providers/api500.rb]: Same issues here

**Review by tmiotto — CHANGES_REQUESTED** [review]

**Review by tmiotto — APPROVED** [review]

**Pull request merged**